spec/unit/shoulda/matchers/active_record/have_rich_text_matcher_spec.rb

```diff
@@ -0,0 +1,137 @@
+          'RichText class' do
+        create_table_posts(has_content_attribute: true)
+    
+        invalid_model = define_model_class(:Post)
+        invalid_record = invalid_model.new(content: "Content")
+    
+        expected_message = 'Expected Post to have :content as a ' \
+          'ActionText::RichText but :content is a String'
+    
+        aggregate_failures do
+          expect(invalid_record).not_to have_rich_text(:content)
+          expect { expect(invalid_record).to have_rich_text(:content) }.
+            to fail_with_message(expected_message)
+        end
+      end
+    
+      it 'fails with the appropriate message when negated' do
+        expect { expect(new_post(has_rich_text: true)).not_to have_rich_text(:content) }.
+          to fail_with_message('Did not expect Post to have a rich text :content')
+      end
+    end
+  elsif active_record_version >= 4 && active_record_version < 5
+    it 'raise a undefined method error when action text is not defined' do
+      create_table_posts
+  
+      invalid_record = -> { new_post(has_rich_text: true) }
+
+      message = /undefined method `has_rich_text' for Post/
+        
+      expect(&invalid_record).to raise_error(
+        NoMethodError,
+        message
+      )
+    end
+
+    it 'does not matches when the provided attribute it\'s not a ' \
+        'RichText class' do
+      create_table_posts(has_content_attribute: true)
+  
+      invalid_model = define_model_class(:Post)
+      invalid_record = invalid_model.new(content: "Content")
+      expected_message = 'Expected Post to have :content as a ' \
+        'ActionText::RichText but :content is a NilClass'
+  
+      aggregate_failures do
+        expect(invalid_record).not_to have_rich_text(:content)
+        expect { expect(invalid_record).to have_rich_text(:content) }.
+          to fail_with_message(expected_message)
+      end
+    end
+  else
+    it 'raise a undefined method error when action text is not defined' do
+      create_table_posts
+  
+      invalid_record = -> { new_post(has_rich_text: true) }
+
+      message = /undefined method `has_rich_text' for Post/
+        
+      expect(&invalid_record).to raise_error(
+        NoMethodError,
+        message
+      )
+    end
+
+    it 'does not matches when the provided attribute it\'s not a ' \
+        'RichText class' do
+      create_table_posts(has_content_attribute: true)
+  
+      invalid_model = define_model_class(:Post)
+      invalid_record = invalid_model.new(content: "Content")
+      expected_message = 'Expected Post to have :content as a ' \
+        'ActionText::RichText but :content is a String'
+  
+      aggregate_failures do
+        expect(invalid_record).not_to have_rich_text(:content)
+        expect { expect(invalid_record).to have_rich_text(:content) }.
+          to fail_with_message(expected_message)
+      end
+    end
+  end
+
+  def create_table_posts(options = {})
```

Comment:
What do you think about using keyword arguments to shorten this a bit, e.g. `def create_table_posts(has_content_attribute: false)`? Same for `new_post`.

Reply:
I like it your suggestion. Is more precise then `options = {}`. Thanks!